### ISSUE TITLE
Anyone has use it in 3090? It seems been slow...

### ISSUE THREAD

**Issue by tpzou** [traveller59/spconv#276]
Anyone has use it in 3090?

**Comment by JohnRomanelis** [comment]
Hi @tpzou! I have installed it, it seems to run just fine. I tested it on the /examples/mnist_sparse.py. I commented out the the if statement along with the print message inside the train function. I used all the default input arguments. Also, i used tqdm to measure the time per epoch and the iteration per sec (processed batches per second).  To complete the first epoch it took about 11sec and it performs ~80 iterations per second.

**Comment by tpzou** [comment]
> Hi @tpzou! I have installed it, it seems to run just fine. I tested it on the /examples/mnist_sparse.py. I commented out the the if statement along with the print message inside the train function. I used all the default input arguments. Also, i used tqdm to measure the time per epoch and the iteration per sec (processed batches per second). To complete the first epoch it took about 11sec and it performs ~80 iterations per second. I also use torch 1.7 with cuda 11.0 and i have tested it with both cuda 11.0 and 11.1 installed separately on ubuntu 20.04.

Thanks for your reply. I didn't directly train it, I use it as a backbone in another released model. It seems much slow than the 2080ti. 
Is your test means that  the spconv is more efficient in 3090? Perhaps I should modify other codes.

**Comment by JohnRomanelis** [comment]
Hey @tpzou , 
I haven't compared it to another GPU unfortunately. I trained it on the aforementioned example so that you can easily replicate the results in case these is something wrong with your installation. Maybe, if you still have access to the 2080ti, you can perform the same test with it!

**Comment by ldtho** [comment]
Hey @tpzou , I can't build spconv on my rtx 3080 with cuda 11, can you please share your cuda cudnn cmake pytorch version with me? It would be really appreciated.

**Comment by tpzou** [comment]
> Hey @tpzou , I can't build spconv on my rtx 3080 with cuda 11, can you please share your cuda cudnn cmake pytorch version with me? It would be really appreciated.

Hi, @ldtho .
My cudnn is 11.0 and the pytorch is 1.7-cudnn11.0, as the pytorch didn't have the cudnn11.1. And the spconv's version is 1.2, if you use spconv 1.0, some codes need to be modified to fit the pytorch version. Another **error** is the **architecture**  should be changed to fit the cudnn11.0 (30 GPU is 8.6, which should be change to 8.0).
Hope it can help you.

**Comment by dimitree54** [comment]
I am also facing pure performance of spconv on 3090.
Running mnist_sparse.py on the machine with pip installed pytorch 1.4 + cuda 10.1 + RTX 2080 GPU (utilization 70-80%):
```
Test set: Average loss: 0.0318, Accuracy: 9903/10000 (99%)

9920512it [02:04, 79751.38it/s]  
```
Running mnist_sparse.py on the machine with compiled pytorch 1.6 + cuda 11.1 + RTX 3090 GPU (utilization 20-30%):
```
Test set: Average loss: 0.0314, Accuracy: 9905/10000 (99%)

9920512it [04:49, 34256.18it/s]  
```

**Comment by tpzou** [comment]
> I am also facing pure performance of spconv on 3090.
> Running mnist_sparse.py on the machine with pip installed pytorch 1.4 + cuda 10.1 + RTX 2080 GPU (utilization 70-80%):
> 
> ```
> Test set: Average loss: 0.0318, Accuracy: 9903/10000 (99%)
> 
> 9920512it [02:04, 79751.38it/s]  
> ```
> 
> Running mnist_sparse.py on the machine with compiled pytorch 1.6 + cuda 11.1 + RTX 3090 GPU (utilization 20-30%):
> 
> ```
> Test set: Average loss: 0.0314, Accuracy: 9905/10000 (99%)
> 
> 9920512it [04:49, 34256.18it/s]  
> ```

Hi, @dimitree54, Which version of spconv did you use?

**Comment by dimitree54** [comment]
@tpzou for both cases I use spconv-1.2.1-cp36-cp36m-linux_x86_64.whl

**Comment by dimitree54** [comment]
I have also tried another setting for 3090:
compiled pytorch 1.8 + cuda 11.1 + RTX 3090 GPU (utilization 35-45%) and it takes 2 min 40 sec (compared to 2 min 54 sec on 2080). This setting is much better than previous, but still I was expecting more from 3090.

On the other hand testing on MNIST is not good benchmark to measure GPU performance because of low number of parameters. I have also tested these configurations on my custom code that uses spconv for high-dimensional task and here is comparison:
pip pytorch 1.4 + cuda 10.1 + RTX 2080 GPU (100% utilization) takes 76 minutes per epoch;
compiled pytorch 1.6 + cuda 11.1 + RTX 3090 GPU (100% utilization) takes 89 minutes per epoch;
compiled pytorch 1.8 + cuda 11.1 + RTX 3090 GPU (100% utilization) takes 61 minutes per epoch.
In all cases I was using compiled spconv 1.2.1.

I was expecting about 3 times performance gain on the new GPU, but it seems that spconv (or maybe pytorch itself) is not optimized for that hardware yet.
